File: Python/SivitasAkademik.py
# import kelas Human
from Human import Human

# deklarasi kelas Sivitas Akademik yang merupakan child dari Human
class SivitasAkademik(Human):

    # privat atribut milik kelas Sivitas Akademik
    __asal_universitas = ""
    __email_edu = ""
    __fakultas = ""

    # method2 public milik kelas Sivitas Akademik

    # constructor
    def __init__(
        self,
        nik=0,
        nama="",
        jenis_kelamin="",
        asal_universitas="",
        email_edu="",
        fakultas="",
    ):
        # constructor super untuk kelas Human
        super().__init__(nik, nama, jenis_kelamin)

        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu
        self.__fakultas = fakultas

    # setter asal universitas
    def setAsalUniversitas(self, asal_universitas):
        self.__asal_universitas = asal_universitas

    # setter email edu
    def setEmailEdu(self, email_edu):
        self.__email_edu = email_edu

    # setter fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

    # getter asal universitas
    def getAsalUniversitas(self):
        return self.__asal_universitas

    # getter email edu
    def getEmailEdu(self):
        return self.__email_edu

    # getter fakultas
    def getFakultas(self):
        return self.__fakultas

File: CPP/SivitasAkademik.cpp
#include "header.hh"

/* Definisi Method Kelas Sivitas Akademik */

// constructor kosong
SivitasAkademik::SivitasAkademik()
    : Human()
{
}

// constructor dengan parameter seluruh atribut Sivitas Akademik dan atribut Humahn
SivitasAkademik::SivitasAkademik(int nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string fakultas)
    : Human(nik, nama, jenis_kelamin)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
    this->fakultas = fakultas;
}

// Set asal universitas
void SivitasAkademik::setAsalUniversitas(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}

// Set email edu
void SivitasAkademik::setEmailEdu(string email_edu)
{
    this->email_edu = email_edu;
}

// Set fakultas
void SivitasAkademik::setFakultas(string fakultas)
{
    this->fakultas = fakultas;
}

// Get asal universitas
string SivitasAkademik::getAsalUniversitas()
{
    return this->asal_universitas;
}

// Get email edu
string SivitasAkademik::getEmailEdu()
{
    return this->email_edu;
}

// Get fakultas
string SivitasAkademik::getFakultas()
{
    return this->fakultas;
}

// destructor
SivitasAkademik::~SivitasAkademik()
{
}
